parquet-column/src/main/java/org/apache/parquet/schema/ColumnOrder.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.schema;
+
+import org.apache.parquet.Preconditions;
+
+/**
+ * Class representing the column order with all the related parameters.
+ */
+public class ColumnOrder {
```

Comment:
Why is this not a single enum? What is the value of having both `ColumnOrder` and `ColumnOrderName`?

Reply:
I followed the design of `parquet-format` and the naming/design of `PrimitiveType` and `PrimitiveTypeName`.  The idea behind this design is to allow additional parameters to be added to `ColumnOrder`. For example defining collations for UTF8 based ordering.
What do you think?

Reply:
+1